core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -149,10 +149,31 @@ public boolean queryStringLenient() {
         return this.queryStringLenient;
     }
 
-    IndicesQueriesRegistry indicesQueriesRegistry() {
+    public IndicesQueriesRegistry indicesQueriesRegistry() {
         return indicesQueriesRegistry;
     }
 
+    public ParsedQuery parse(QueryBuilder<?> queryBuilder) {
+        QueryShardContext context = cache.get();
+        context.reset();
+        context.parseFieldMatcher(parseFieldMatcher);
+        try {
+            return innerParse(context, queryBuilder);
+        } catch (ParsingException e) {
```

Comment:
should it be `QueryShardException`? or `ParsingException | QueryShardException` ?